sn_node/src/node/bootstrap/join.rs

```diff
@@ -375,6 +369,51 @@ impl<'a> Joiner<'a> {
         }
     }
 
+    // The retry responses may containing different expected ages, or only having
+    // more than 1/3 of elders voted (which blocks the aggregation).
+    // So a restart with new name shall happen when: received 1/3 of elders voted
+    // And with an age of: the most expected age
+    fn insert_retry_response(
+        &mut self,
+        sender: Peer,
+        expected_age: u8,
+        elders: BTreeSet<Peer>,
+    ) -> Option<u8> {
+        if !elders.contains(&sender) {
+            error!("Sender {sender:?} of the retry-response is not part of the elders {elders:?}");
+            return None;
+        }
+        let _ = self.retry_responses_cache.insert(sender, expected_age);
+        let mut responded = 0;
+        let mut expected_ages: BTreeMap<u8, u8> = Default::default();
+        for (elder, expected_age) in self.retry_responses_cache.iter() {
+            if elders.contains(elder) {
+                responded += 1;
+                *expected_ages.entry(*expected_age).or_insert(1) += 1;
```

Comment:
You are double counting the first entry
```suggestion
                *expected_ages.entry(*expected_age).or_insert(0) += 1;
```